Mutect1

buehlere opened the pull request.

ErikDanielsson committed:
Add fastp and remove fastqc

ErikDanielsson committed:
Remove input sections

mirpedrol committed:
restructure modules folder

mirpedrol committed:
mimic old sha eb4bc244de7eaef8e8ff0d451e4ca2e4b2c29821

mirpedrol committed:
mimic las commit

mashehu committed:
Merge branch 'branch-tester-restructure' into 'branch-tester'

Branch tester restructure

See merge request nf-core/modules-test!2

mashehu committed:
Update .nf-core.yml

mirpedrol committed:
add a new org_path other than nf-core

mirpedrol committed:
add test files

mirpedrol committed:
add pytest_modules.yml

mirpedrol committed:
fix path

mirpedrol committed:
fix fastp name :(

mirpedrol committed:
fix relative paths

mirpedrol committed:
add nextflow.config

mirpedrol committed:
add nextflow.config

mirpedrol committed:
add bam_stats_samtools

mirpedrol committed:
add samtools

buehlere commented:
Another side note, is both `master` and `develop` are syncing with `nf-core/master` on a daily basis. I'm currently updating the documentation to reflect this!

anoronh4 commented:
I wonder if it makes sense to use github actions to operationalize the testing. I opened PR #30 to make it so all the checks that run on a PR against master can also be run on a PR against develop.

buehlere commented:
> I wonder if it makes sense to use github actions to operationalize the testing. I opened PR #30 to make it so all the checks that run on a PR against master can also be run on a PR against develop.

This is a great idea! I added a comment on the PR so let me know your thoughts.

buehlere commented:
> Instead of using install_data.sh, i wonder if we can use smaller data, or instead use lfs (large file storage) on the enterprise github? Also, how come we can't use the fastqs [here](https://github.com/nf-core/test-datasets/tree/modules)?

We might be able to create a smaller one, but this is just the one we had on hand. Unfortunately, I already know it's too large to upload to enterprise github. I'm not too familiar with nf-core test-datasets, but am definitely interested in exploring other options as I agree this is far from the cleanest solution.

huyu335 commented:
> > Instead of using install_data.sh, i wonder if we can use smaller data, or instead use lfs (large file storage) on the enterprise github? Also, how come we can't use the fastqs [here](https://github.com/nf-core/test-datasets/tree/modules)?
> 
> We might be able to create a smaller one, but this is just the one we had on hand. Unfortunately, I already know it's too large to upload to enterprise github. I'm not too familiar with nf-core test-datasets, but am definitely interested in exploring other options as I agree this is far from the cleanest solution.

How about adding the test data to a image and upload to the package of this organization?

huyu335 commented:
On local, it looks like `nf-core subworkflows test extractumi` and `pytest --tag subworkflows/extractumi --symlink --kwdof --git-aware --color=yes` were not working for me. They took 2 hours and failed without clear error(probably out of memory or timeout). But `nextflow run ./tests/subworkflows/nf-core/extractumi -entry test_extractumi -c ./tests/config/nextflow.config` worked.

buehlere commented:
> On local, it looks like `nf-core subworkflows test extractumi` and `pytest --tag subworkflows/extractumi --symlink --kwdof --git-aware --color=yes` were not working for me. They took 2 hours and failed without clear error(probably out of memory or timeout). But `nextflow run ./tests/subworkflows/nf-core/extractumi -entry test_extractumi -c ./tests/config/nextflow.config` worked.

Yes I noticed this as well and haven't been able to get it to work locally. The pytest seems to work fine on git actions and direct run of the nextflow test seems to work as well so not exactly sure what the issue is. I think it might be a bug coming from nf-core? I've only experienced this with subworkflows. @anoronh4 I think you observed this issue as well?

buehlere commented:
try to remove all of `install_data.sh` function.

buehlere commented:
blank test

buehlere commented:
I used these commands to help with cleanup: 
`git checkout develop modules/nf-core/bcftools/`
`git checkout develop modules/nf-core/tabix/`
`git checkout develop subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/`
`git checkout develop tests/subworkflows/nf-core/vcf_annotate_ensemblvep_snpeff/`
`git checkout develop tests/modules/nf-core/tabix/`
`git checkout develop tests/modules/nf-core/bcftools/`
`git checkout develop tests/config/pytest_modules.yml`
and then did a full merge with `develop`

Commit:
init

Commit:
working on stub test

Commit:
adjusting stub test

Commit:
adding basic testing

buehlere committed:
Merge branch 'develop' into feature/gbcms

Commit:
Update nf-test.yml

Commit:
Update main.nf

Commit:
Update main.nf

Commit:
using wrong test.yml, removing and correcting

Commit:
Merge remote-tracking branch 'origin/develop' into feature/gbcms

Commit:
delete old test yml

Commit:
prettier

Commit:
updating module inputs and meta.yml

Commit:
fixing up test

Commit:
Update meta.yml

Commit:
fix linting, docs and outputs

Commit:
resolve merge

Commit:
fixing docs

Commit:
Update meta.yml

Commit:
more test refactoring

Commit:
Merge branch 'feature/gbcms' of https://github.com/mskcc-omics-workflows/modules into feature/gbcms

Commit:
fixing up tests and inputs from code review

Commit:
fix test output

Commit:
fix outputs

Commit:
fix outputs retry

Commit:
get rid of old params

Commit:
dropping params in nextflow.config

Commit:
fixing verion name

Commit:
fixing verion name retry

Commit:
removing conda test for gbcms

Commit:
removing conda test for gbcms

Commit:
removing conda test for gbcms

Commit:
adjusting container variable and undoing changes to test.yml

Commit:
adjusting script to deal with conda case

Commit:
gbcms

Commit:
fixing exclude tags

Commit:
Update test.yml

Commit:
remove old argument, config spacing

Commit:
Update test.yml

Commit:
ignore DS_Store

Commit:
basic nf-test for gbcms

Commit:
changing tag

Commit:
upgrading test

Commit:
re-formatting

Commit:
master -> main

Commit:
foramatting and remove old tests

Commit:
remove gbcms from pytest.yml

Commit:
updates for linting

Commit:
adjusting for conda test

Commit:
ignoring conda test

Commit:
removing conda

Commit:
updating stub and removing old comments

:heavy_check_mark: cleaning up stub test and old comments

Commit:
Update main.nf

Commit:
updating test.yml

we want the tests to run on PRs to main and develop

buehlere committed:
Update test.yml

✔️ adding back conda test

Commit:
removing conda test, updating config

:heavy_check_mark: from review on #10, updating nextflow config and ignoring conda test on module

Commit:
remove user emulation

Commit:
remove nf-core modules

Commit:
init setup
